### PR TITLE
Additional Preprocessing

### PR DESCRIPTION
**Pull request by johannzapf**
This PR implements normalization and encoding.

There are two new files:
**preprocessing_one_hot_encoding.py** contains methods to generate datasets with one hot encoding. There are options to encode only ordinal columns with more than yes/no options or all of them. The target column can also be one hot encoded via a function parameter.
**preprocessing_label_encoding.py** contains methods for datasets with label encoding (i.e. with the values already in the ordinal columns).

Both new files also contain methods for over/undersampled variants of the respective datasets.

The file **preprocessing.py** now just contains some helper methods to load the dataset, remove / rename columns and to normalize the numerical values (i.e. the steps that need to be taken for all preprocessing variants).
It now also contains a method **get_train_test_split()** which can be used in the notebooks to generate train test splits from the chosen preprocessing approach

### CODE REVIEW

**Comment by Niklas-23** [comment]
@johannzapf I could fix the pipeline issue. The problem was that the pipeline used the small example dataset and therefore one class had fewer tuples left after preprocessing. I extended the dataset example and now everything works fine.

File: test/test_preprocessing.py [hunk 1]
```diff
@@ -1,10 +1,11 @@
-from preprocessing.preprocessing import get_preprocessed_brfss_train_test_split
+from preprocessing.preprocessing import get_train_test_split, get_preprocessed_brfss_dataset
 
 
 class TestPreprocessing:
 
     def test_get_preprocessed_train_test_split(self):
-        train_data, test_data, train_target, test_target = get_preprocessed_brfss_train_test_split()
+        dataset, target = get_preprocessed_brfss_dataset()
+        train_data, test_data, train_target, test_target = get_train_test_split(dataset, target)
```
Comment (johannzapf) [inline test/test_preprocessing.py:8]: If you previously used the get_preprocessed_brfss_train_test_split() method, you must now first call the preprocessing method you want to use (i.e. which encoding and sampling variant) and then call the preprocessing.get_train_test_split() function.

**Review by dmonti9 — COMMENTED** [review]
Niceee, great work, really like the new structure! And thanks for the detailed documentation!

File: preprocessing/preproccessing_one_hot_encoding.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import pandas as pd
+from pandas import DataFrame
+from typing import Tuple
+
+from sklearn.preprocessing import OneHotEncoder
+
+from preprocessing import get_preprocessed_brfss_dataset, diabetes_columns, oversample_dataset, \
+    undersample_dataset
```
Comment (johannzapf) [inline preprocessing/preproccessing_one_hot_encoding.py:8]: PyCharm underlines this import red, but it is the only way I could make it work when I actually run the code. Any ideas @Niklas-23 ?

File: preprocessing/preprocessing_label_encoding.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+from pandas import DataFrame
+from typing import Tuple
+
+from preprocessing import get_preprocessed_brfss_dataset, oversample_dataset, undersample_dataset
```
Comment (johannzapf) [inline preprocessing/preprocessing_label_encoding.py:4]: Same thing as above here

File: preprocessing/preproccessing_one_hot_encoding.py (hunk 2 above)
Comment (Niklas-23) [reply preprocessing/preproccessing_one_hot_encoding.py]: I think this must be a problem with your local setup, I don't understand how you can run the code when PyCharm underlines this as wrong for you too. I added a unit test showing the error.

**Review by Niklas-23 — APPROVED** [review]
Lgtm now. Thanks :)
